### ISSUE TITLE
Wormhole Usage

### ISSUE THREAD

**Issue by InfamousChloe** [ensingm2/SteamMonsterGameScript#161]
Currently, wormholes are dropping as soon as it's available on each 100th level, but this seems incredibly inefficient compared to saving them for the end game.

**Comment by cdmichaelb** [comment]
Too tired to stick around for a fix, but I imagine in the long run as long as you're in one of the high pop rooms, it wont be that big of a deal since we got the new damage ranks.

**Comment by milkfat** [comment]
Wormhole should be used in conjunction with Raining Gold rounds divisible by 100. Wormholes are activated _after_ the level is completed. They will continue to stack until the level is completed.

**Comment by ensingm2** [comment]
@InfamousMyzt Are you sure? I didn't write the code but looking at the condition:

`if (((getEstimatedLevelsLeft() % 100) < getAbilityItemQuantity(ABILITIES.WORMHOLE) && lvl % 100 === 0) || hasTimeLeftToUseConsumable(ABILITIES.WORMHOLE))`

it seems about right. use it if:
- you're running out of time in the round and need to spam

OR
- the number of estimated 'Raining Gold' boss rounds is under your wormhole count, AND you're on a 'Raining Gold' round.

I can't verify since I don't have a wormhole and am not in a top room, but @milkfat's comment seems to support this.

Is this still an issue, or should I close? I'm not super familiar with wormhole strategy.

**Comment by InfamousChloe** [comment]
I'm not sure now, the script burned all my wormholes at the beginning. They should be used every 100 rounds but only at the end of the game, it shouldn't skip early levels. I would say they shouldn't be used until the last hour, even if you have 60+. During the last hour there are a ton of cooldown resets being blown.

**Comment by Kaboom108** [comment]
It's using my wormholes currently, in game 43138.  Expected level is 432050, current level is 88920.  I have 24 wormholes left.  It started the last time I updated.  Disabling consumable use seems to stop it.  Would be nice to have a setting to just disable wormhole use for people that want to keep them for now.

**Comment by thelulzy** [comment]
We should probably sync up with the wormhole strat update that wchill's and steamDB's just rolled out

**Comment by EraYaN** [comment]
Which is each 500, you can change your userscript by changing both 100's to 500's. And add 

``` javascript
&& lvl > 100000 // or whatever
```

**Comment by vanZeben** [comment]
I was in a super low level room that wasn't progressing worth crap (literally 27 seconds a level -.-), so I couldnt fully test it out, especially since it isnt end of day, but it wasnt using wormholes for me in my piddly room with a prospected level of 3k

**Comment by ensingm2** [comment]
Think I mostly fixed things on my local copy, just waiting to test.

...of course I'm stuck outside of game and can't actually test aynthing...

Waiting on players is fun.

**Comment by vanZeben** [comment]
SAME! What room are you in!

**Comment by ensingm2** [comment]
I just JoinGame(0)'d. Looks like I'm in 43755

**Comment by vanZeben** [comment]
aweee, im in 43758 xD

**Comment by ensingm2** [comment]
Can someone look through https://github.com/ensingm2/SteamMonsterGameScript/commit/d608a941e7ce626eff5b92a72503ea73346fbcc4 and make sure I didn't fuck anything up?

Seems correct to me, but kind of tough to test things out.

**Comment by vanZeben** [comment]
Looks right as far as I can see
